python/pyspark/sql/catalog.py

```diff
@@ -123,19 +132,52 @@ def listCatalogs(self) -> List[CatalogMetadata]:
             catalogs.append(CatalogMetadata(name=jcatalog.name, description=jcatalog.description))
         return catalogs
 
-    @since(2.0)
```

Comment:
why do we remove `since`?

Reply:
Because it breaks the docstring format. `since` adds the `versionadded` directive at the end of docstring which does not work in NumPy documentation style when `Parameters`, etc are specified. We should remove these all eventually.